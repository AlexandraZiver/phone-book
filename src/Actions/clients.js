import { createAction } from "@reduxjs/toolkit";

import { GET_CLIENTS } from "../constants/actionsTypes";

const getClientsAction = createAction(GET_CLIENTS);
export default getClientsAction;
