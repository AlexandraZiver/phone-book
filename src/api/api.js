import axios from "axios";

import { BASE_URL } from "../constants/endpoints";

const axiosInstance = axios.create({
  baseURL: `${BASE_URL}/clients`,
});

class ClientService {
  static async getAll() {
    const { data } = await axiosInstance.get();
    return data;
  }
  static async getById(id) {
    const { data } = await axiosInstance.get(id);
    return data;
  }
}

export default ClientService;
