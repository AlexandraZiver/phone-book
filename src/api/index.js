export { default as ClientService } from "../store/clients/clients.actions";
