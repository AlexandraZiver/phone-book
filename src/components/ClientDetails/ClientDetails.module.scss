@import "../../styles/theme";
<<<<<<< HEAD
@import "../../styles/mixins/header.scss";
=======
@import "../../styles/mixins/background.scss";
>>>>>>> 3233376 (feature/clients-details)

@mixin column {
  display: flex;
  flex-direction: column;
}

.WrapperClientDetails {
  height: 100%;
<<<<<<< HEAD
  background-image: linear-gradient(theme-color(dark), theme-color(dark));
  box-sizing: border-box;
  padding: 1em;
  backdrop-filter: blur(10px);
}
.Container {
  @include column;
  gap: 0.5em;
=======
  @include background($gradient: true);
  box-sizing: border-box;
  padding: 1em;
>>>>>>> 3233376 (feature/clients-details)
}

.Name {
  font-size: 1.5em;
<<<<<<< HEAD
=======
  font-weight: 600;
>>>>>>> 3233376 (feature/clients-details)
}

.Image {
  width: 150px;
  height: 150px;
  margin: 1em 1em 1em 0;
  flex-shrink: 0;
}

.InfoClient {
  display: flex;
<<<<<<< HEAD
  flex-direction: column;
}

.Job {
  color: theme-color(secondary);
  padding: 1em 1em 1em 0;
  text-transform: uppercase;
}

.Info {
  display: flex;
  align-items: center;
=======
  align-items: center;
}

.Job {
  padding: 1em;
  background-color: theme-color(secondary);
  margin: 1em;
  border-radius: 0.5em;
}

.Info {
  @include column;
  gap: 0.5em;
>>>>>>> 3233376 (feature/clients-details)
}

.Icon {
  color: theme-color(secondary);
}

.Address {
  @include column;
}
<<<<<<< HEAD
=======

.BtnReturn {
  border: 1px solid theme-color(secondary);
  border-radius: 0.5em;
  padding: 0.5em;
  background-color: theme-color(secondary);
  margin: 1em 1em 1em 0;
}

.BtnReturn:hover {
  background-color: theme-color(primary);
}
>>>>>>> 3233376 (feature/clients-details)
