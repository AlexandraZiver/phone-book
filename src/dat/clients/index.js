import clientReducer from "./clients.reducer";

export { getClientsAction } from "./clients.reducer";
export default clientReducer;
