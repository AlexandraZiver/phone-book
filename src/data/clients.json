[
  {
    "general": {
      "firstName": "Liana",
      "lastName": "Crooks",
      "avatar": ""
    },
    "job": {
      "company": "Ledner, Johnson and Predovic",
      "title": "Investor Functionality Coordinator"
    },
    "contact": {
      "email": "Gerry_Hackett77@gmail.com",
      "phone": "(895) 984-0132"
    },
    "address": {
      "street": "1520 Zemlak Cove",
      "city": "New Devon",
      "zipCode": "42586-7898",
      "country": "Guinea-Bissau"
    }
  },
  {
    "general": {
      "firstName": "Deontae",
      "lastName": "Dare",
      "avatar": ""
    },
    "job": {
      "company": "D'Amore, Dicki and Borer",
      "title": "International Applications Consultant"
    },
    "contact": {
      "email": "Kellie.Marvin38@yahoo.com",
      "phone": "1-615-843-3426 x600"
    },
    "address": {
      "street": "65901 Glover Terrace",
      "city": "Alden ton",
      "zipCode": "57744-4248",
      "country": "Kenya"
    }
  },
  {
    "general": {
      "firstName": "Cortez",
      "lastName": "",
      "avatar": ""
    },
    "job": {
      "company": "McKenzie Group",
      "title": "Forward Branding Developer"
    },
    "contact": {
      "email": "Sage_Wiegand@hotmail.com",
      "phone": "725.583.0926 x0430"
    },
    "address": {
      "street": "376 Reginald Dam",
      "city": "Port Enid fort",
      "zipCode": "51294-8361",
      "country": "Belarus"
    }
  },
  {
    "general": {
      "firstName": "Geoffrey",
      "lastName": "Russel",
      "avatar": "https://randomuser.me/api/portraits/men/9656ybdfgf7.jpg"
    },
    "job": {
      "company": "Nienow and Sons",
      "title": "Central Intranet Designer"
    },
    "contact": {
      "email": "Daron.Bartoletti9@gmail.com",
      "phone": "646.580.9390"
    },
    "address": {
      "street": "5050 Iva Extensions",
      "city": "Madonna burgh",
      "zipCode": "74470-6362",
      "country": "Fiji"
    }
  },
  {
    "general": {
      "firstName": "Christian",
      "lastName": "Wuckert",
      "avatar": "https://randomuser.me/api/portraits/men/36.jpg"
    },
    "job": {
      "company": "Jakubowski Inc",
      "title": "Future Branding Assistant"
    },
    "contact": {
      "email": "Zechariah48@hotmail.com",
      "phone": "555-516-5564"
    },
    "address": {
      "street": "1946 Nolan Mountain",
      "city": "Garnet stad",
      "zipCode": "79438",
      "country": "Puerto Rico"
    }
  },
  {
    "general": {
      "firstName": "Joana",
      "lastName": "Breitenberg",
      "avatar": "https://images.unsplash.com/photo-1511485977113-f34c92461ad9?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&fit=crop&h=200&w=200&ixid=eyJhcHBfaWQiOjE3Nzg0fQ"
    },
    "job": {
      "company": "Jacobson - Fay",
      "title": "Global Factors Officer"
    },
    "contact": {
      "email": "Jaylon92@hotmail.com",
      "phone": "202.387.0215 x7568"
    },
    "address": {
      "street": "3446 Isabelle Shore",
      "city": "Port Kayli",
      "zipCode": "63713-9923",
      "country": "Switzerland"
    }
  },
  {
    "general": {
      "firstName": "Elton",
      "lastName": "Pfannerstill",
      "avatar": "https://randomuser.me/api/portraits/men/16.jpg"
    },
    "job": {
      "company": "Franecki LLC",
      "title": "Product Applications Assistant"
    },
    "contact": {
      "email": "Tomasa26@hotmail.com",
      "phone": "168.457.7936 x4319"
    },
    "address": {
      "street": "1975 Creola Streets",
      "city": "South Favian",
      "zipCode": "65666-6266",
      "country": "Afghanistan"
    }
  },
  {
    "general": {
      "firstName": "Alvena",
      "lastName": "Paucek",
      "avatar": "https://randomuser.me/api/portraits/women/30.jpg"
    },
    "job": {
      "company": "Goldner - Haag",
      "title": "Corporate Interactions Assistant"
    },
    "contact": {
      "email": "Oran66@yahoo.com",
      "phone": "(754) 491-0343 x6060"
    },
    "address": {
      "street": "95820 Bud Trail",
      "city": "West Randy furt",
      "zipCode": "98923",
      "country": "French Polynesia"
    }
  },
  {
    "general": {
      "firstName": "Lew",
      "lastName": "Daniel",
      "avatar": "https://randomuser.me/api/portraits/men/22.jpg"
    },
    "job": {
      "company": "Corwin LLC",
      "title": "Product Research Liason"
    },
    "contact": {
      "email": "Cordelia.Bartoletti28@gmail.com",
      "phone": "142.115.7141 x943"
    },
    "address": {
      "street": "04363 Torphy Club",
      "city": "East Heidi",
      "zipCode": "10926-2413",
      "country": "Estonia"
    }
  },
  {
    "general": {
      "firstName": "Darlene",
      "lastName": "Davis",
      "avatar": "https://images.unsplash.com/photo-1510227272981-87123e259b17?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&fit=crop&h=200&w=200&s=3759e09a5b9fbe53088b23c615b6312e"
    },
    "job": {
      "company": "Franecki and Sons",
      "title": "Internal Functionality Supervisor"
    },
    "contact": {
      "email": "Erich36@gmail.com",
      "phone": "(546) 314-2504"
    },
    "address": {
      "street": "1513 Kessler Crossing",
      "city": "South Randi fort",
      "zipCode": "60194",
      "country": "Malta"
    }
  },
  {
    "general": {
      "firstName": "Savannah",
      "lastName": "Predovic",
      "avatar": "https://images.unsplash.com/photo-1520810627419-35e362c5dc07?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&fit=crop&h=200&w=200&ixid=eyJhcHBfaWQiOjE3Nzg0fQ"
    },
    "job": {
      "company": "O'Connell - Koepp",
      "title": "International Integration Liason"
    },
    "contact": {
      "email": "Torey_Stroman@hotmail.com",
      "phone": "324.808.6122"
    },
    "address": {
      "street": "660 Lueilwitz Island",
      "city": "East Okey port",
      "zipCode": "26277",
      "country": "Sudan"
    }
  },
  {
    "general": {
      "firstName": "Nicolette",
      "lastName": "Rogahn",
      "avatar": "https://randomuser.me/api/portraits/women/79.jpg"
    },
    "job": {
      "company": "Zemlak - Larkin",
      "title": "Internal Group Agent"
    },
    "contact": {
      "email": "Rowena.Kemmer93@yahoo.com",
      "phone": "579-666-9685"
    },
    "address": {
      "street": "6622 Kaitlin Drive",
      "city": "New Israel",
      "zipCode": "07116",
      "country": "Turkmenistan"
    }
  },
  {
    "general": {
      "firstName": "Aidan",
      "lastName": "Stracke",
      "avatar": "https://randomuser.me/api/portraits/men/1.jpg"
    },
    "job": {
      "company": "Denesik, Dicki and Schmitt",
      "title": "National Accounts Officer"
    },
    "contact": {
      "email": "Chelsey.Powlowski38@hotmail.com",
      "phone": "689.859.2512 x81508"
    },
    "address": {
      "street": "498 King Track",
      "city": "Toy fort",
      "zipCode": "07905-5925",
      "country": "Latvia"
    }
  },
  {
    "general": {
      "firstName": "Tristin",
      "lastName": "Eichmann",
      "avatar": "https://randomuser.me/api/portraits/men/52.jpg"
    },
    "job": {
      "company": "Marvin, Pfannerstill and Braun",
      "title": "Central Communications Consultant"
    },
    "contact": {
      "email": "Marisa_Miller54@yahoo.com",
      "phone": "335-788-4534"
    },
    "address": {
      "street": "2686 Ebert Parks",
      "city": "West Lexus",
      "zipCode": "70293-4149",
      "country": "Brazil"
    }
  },
  {
    "general": {
      "firstName": "Malika",
      "lastName": "Feeney",
      "avatar": "https://randomuser.me/api/portraits/women/87.jpg"
    },
    "job": {
      "company": "Pfannerstill - Christiansen",
      "title": "Customer Assurance Designer"
    },
    "contact": {
      "email": "Stone15@yahoo.com",
      "phone": "578-268-2680 x495"
    },
    "address": {
      "street": "415 Homenick Lodge",
      "city": "North Nelson borough",
      "zipCode": "05142",
      "country": "Netherlands"
    }
  },
  {
    "general": {
      "firstName": "Ross",
      "lastName": "Dickens",
      "avatar": "https://randomuser.me/api/portraits/men/46.jpg"
    },
    "job": {
      "company": "Champlin, Leannon and O'Connell",
      "title": "Legacy Marketing Coordinator"
    },
    "contact": {
      "email": "Scottie.Swift@hotmail.com",
      "phone": "165-084-3752 x336"
    },
    "address": {
      "street": "98494 Clemens Oval",
      "city": "Heller view",
      "zipCode": "58090",
      "country": "Mozambique"
    }
  },
  {
    "general": {
      "firstName": "Granville",
      "lastName": "Larson",
      "avatar": "https://randomuser.me/api/portraits/women/33.jpg"
    },
    "job": {
      "company": "Metz - Bashirian",
      "title": "Senior Brand Associate"
    },
    "contact": {
      "email": "Judd6@hotmail.com",
      "phone": "1-817-435-1136 x8014"
    },
    "address": {
      "street": "357 Jeffrey Avenue",
      "city": "Hand side",
      "zipCode": "89647-5238",
      "country": "Morocco"
    }
  },
  {
    "general": {
      "firstName": "Donnie",
      "lastName": "Macejkovic",
      "avatar": "https://randomuser.me/api/portraits/women/63.jpg"
    },
    "job": {
      "company": "Schmeler - Romaguera",
      "title": "Senior Functionality Facilitator"
    },
    "contact": {
      "email": "Camryn_Gerhold@hotmail.com",
      "phone": "(186) 005-2043"
    },
    "address": {
      "street": "76139 Hayes Plaza",
      "city": "Emard stad",
      "zipCode": "45180",
      "country": "Western Sahara"
    }
  },
  {
    "general": {
      "firstName": "Estell",
      "lastName": "Baumbach",
      "avatar": "https://randomuser.me/api/portraits/men/12.jpg"
    },
    "job": {
      "company": "Kuphal - Renner",
      "title": "Principal Mobility Associate"
    },
    "contact": {
      "email": "Cassie_Brekke@yahoo.com",
      "phone": "1-354-996-2400 x08787"
    },
    "address": {
      "street": "64955 Ottilie Port",
      "city": "Bartell mouth",
      "zipCode": "62822-8781",
      "country": "Belgium"
    }
  },
  {
    "general": {
      "firstName": "Amelie",
      "lastName": "Bradtke",
      "avatar": "https://images.unsplash.com/photo-1438761681033-6461ffad8d80?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&fit=crop&h=200&w=200&s=046c29138c1335ef8edee7daf521ba50"
    },
    "job": {
      "company": "Swift LLC",
      "title": "Dynamic Program Representative"
    },
    "contact": {
      "email": "Prudence.Hane49@yahoo.com",
      "phone": "699-872-8424"
    },
    "address": {
      "street": "198 White Isle",
      "city": "Breitenberg land",
      "zipCode": "02161",
      "country": "Montserrat"
    }
  },
  {
    "general": {
      "firstName": "Elmer",
      "lastName": "D'Amore",
      "avatar": "https://randomuser.me/api/portraits/men/32.jpg"
    },
    "job": {
      "company": "Schaden Group",
      "title": "Regional Brand Strategist"
    },
    "contact": {
      "email": "Margarett57@gmail.com",
      "phone": "1-236-341-6098 x2838"
    },
    "address": {
      "street": "26952 Welch Course",
      "city": "Lake Carmella land",
      "zipCode": "85577-5136",
      "country": "Israel"
    }
  }
]
