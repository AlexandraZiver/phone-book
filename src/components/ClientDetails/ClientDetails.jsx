import React from "react";
import { useParams } from "react-router-dom";
import { List } from "semantic-ui-react";

import { Size } from "../../constants/size";
import { clientsApi } from "../../store/clients/clientsApi";
import { Avatar } from "../Icon";
import LoadingAndError from "../LoadingAndError";
import style from "./ClientDetails.module.scss";

const ClientDetails = () => {
  const { id } = useParams();

  const { data: client, error, status } = clientsApi.useFetchClientByIdQuery(id);

  return (
    <div className={style.WrapperClientDetails}>
      <LoadingAndError hasData={!!client} status={status} error={error} size={Size.BIG} includeText>
        <List className={style.Container}>
          <List.Content className={style.Info}>
            <Avatar
              avatar={client?.general.avatar}
              firstName={client?.general.firstName}
              lastName={client?.general.lastName}
              size="large"
            />
            <List.Content className={style.InfoClient}>
              <div className={style.Name}>
                {client?.general.firstName} {client?.general.lastName}
              </div>
              <div className={style.Job}>{client?.job.title}</div>
            </List.Content>
          </List.Content>
          <List.Item>
            <List.Icon className={style.Icon} name="users" />
            <List.Content>{client?.job.company}</List.Content>
          </List.Item>
          <List.Item>
            <List.Icon className={style.Icon} name="mail" />
            <List.Content>
              <a href={client?.contact.email}>{client?.contact.email}</a>
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content>
              <List.Icon className={style.Icon} name="phone" />
              {client?.contact.phone}
            </List.Content>
          </List.Item>
          <List.Content>
            <List.Icon className={style.Icon} name="map" />
            {client?.address.zipCode}
          </List.Content>
          <List.Item>
            <List.Icon className={style.Icon} name="marker" />
            <List.Content>{client?.address.city}</List.Content>
          </List.Item>
          <List.Content>
            <List.Icon className={style.Icon} name="map marker alternate" />
            {client?.address.country}
            {client?.address.street}
          </List.Content>
        </List>
      </LoadingAndError>
    </div>
  );
};

export default ClientDetails;
