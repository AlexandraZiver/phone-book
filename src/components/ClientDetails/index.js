import ClientDetails from "./ClientDetails";

export default ClientDetails;
