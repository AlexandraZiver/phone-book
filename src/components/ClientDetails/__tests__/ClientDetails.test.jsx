import { screen, render } from "@testing-library/react";
import ClientDetails from "../index";
import { useParams } from "react-router-dom";
import { clientsApi } from "../../../store/api/clients";
jest.mock("../../../store/api/clients", () => ({
  clientsApi: {
    useFetchClientByIdQuery: jest.fn(),
  },
}));

jest.mock("semantic-ui-react", () => {
  const React = require("react");
  return {
    List: jest.fn().mockImplementation(() => <div></div>),
  };
});
jest.mock("react-router-dom", () => ({
  useParams: jest.fn(),
}));

describe("ClientDetails component", () => {
  beforeEach(() => {
    useParams.mockClear();
  });
  it("should render client details", () => {
    const mockClient = {
      general: {
        firstName: "John",
        lastName: "Doe",
        avatar: "https://example.com/avatar.jpg",
      },
      job: {
        title: "Software Engineer",
        company: "Example Inc.",
      },
      contact: {
        email: "john.doe@example.com",
        phone: "555-1234",
      },
      address: {
        street: "123 Main St",
        city: "Anytown",
        zipCode: "12345",
        country: "USA",
      },
      id: 0,
    };

    clientsApi.useFetchClientByIdQuery.mockReturnValue({
      data: mockClient,
      error: null,
      status: "fulfilled",
    });

    useParams.mockReturnValue({ id: "0" });

    const { container, getByText } = render(<ClientDetails />);
    const ClientDetailsElement = container.querySelector(".WrapperClientDetails");
    expect(ClientDetailsElement).toBeInTheDocument();
    expect(ClientDetailsElement).toMatchSnapshot();
  });
});
