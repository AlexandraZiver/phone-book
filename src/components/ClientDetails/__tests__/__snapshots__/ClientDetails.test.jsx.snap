// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientDetails component should render client details 1`] = `
<div
  class="WrapperClientDetails"
>
  <div />
</div>
`;
